examples/hubert/dataset/hubert_dataset.py

```diff
@@ -0,0 +1,330 @@
+        """Get the list of paths for iteration.
+        Args:
+            tsv_dir (Path): The root directory of the ``.tsv`` file list.
+            dataset (str): The dataset for training. Options: [``librispeech``, ``librilight``].
+            subset (str): The subset of the dataset. Options: [``train``, ``valid``].
+            min_sample (int): The minimum number of audio samples in the dataset.
+            max_sample (int): The maximum number of audio samples in the dataset.
+
+        Returns:
+            (numpy.array) List of file paths.
+            (numpy.array) List of indices that qualify ``min_sample`` <= length <= ``max_sample``.
+            (numpy.array) List of waveform lengths.
+        """
+        f_ind_len_list = []
+        with open(tsv_dir / f"{dataset}_{subset}.tsv") as f:
+            root = f.readline().rstrip()
+            for index, line in enumerate(f):
+                path, nsample = line.split("\t")
+                path = f"{root}/{path}"
+                nsample = int(nsample)
+                if min_sample <= nsample <= max_sample:
+                    f_ind_len_list.append((path, index, nsample))
+        f_ind_len_list.sort(key=lambda x: x[2])  # sort the file lists by the sequence length
+        f_list, ind_list, len_list = [], [], []
+        for ele in f_ind_len_list:
+            f_list.append(ele[0])
+            ind_list.append(ele[1])
+            len_list.append(ele[2])
+        return np.asarray(f_list), np.asarray(ind_list), np.asarray(len_list)
+
+    def _load_audio(
+        self,
+        index: int
+    ) -> Tensor:
+        """Load waveform given the sample index of the dataset.
+        Args:
+            index (int): The sample index.
+
+        Returns:
+            (Tensor): The corresponding waveform Tensor.
+        """
+        wav_path = self.f_list[index]
+        waveform, sample_rate = torchaudio.load(wav_path)
+        assert waveform.shape[1] == self.len_list[index]
+        return waveform
+
+    def _load_labels(
+        self,
+        label_dir: Path,
+        dataset: str,
+        subset: str
+    ) -> np.array:
+        """Load all labels to memory into a numpy array.
+        Args:
+            label_dir (Path): The directory that contains the label file.
+            dataset (str): The dataset for training. Options: [``librispeech``, ``librilight``].
+            subset (str): The subset of the dataset. Options: [``train``, ``valid``].
+
+        Returns:
+            (np.array): The numpy arrary that contains the labels for each audio file.
+        """
+        with open(label_dir / f"{dataset}_{subset}.pt") as f:
+            labels = [line.rstrip() for line in f]
+            labels = [[int(ele) for ele in labels[i].split()] for i in self.ind_list]
+        return np.asarray(labels, dtype=object)
+
+    def __getitem__(self, index):
+        waveform = self._load_audio(index)
+        length = waveform.shape[1]
+        label = torch.Tensor(self.labels[index])
```

Comment:
BTW, pytorch prefers using `torch.tensor()` to create Tensor rather than `torch.Tensor`